### ISSUE TITLE
Consider adding compile-definition "FL_DLL" to MSVC 2022 builds with shared libraries

### ISSUE THREAD

**Issue by ggarra13** [fltk/fltk#960]
[Title changed by @Albrecht-S, see below.]

**Describe the bug**
At least 20 symbols are not exported from FLTK when building with RelWithDebInfo.

**To Reproduce**
Self contained build example attached.  Read README.txt.

**Expected behavior**
Linking should not fail and there should be an installed program in install/bin/mrv2.exe that pops up a Window with a button.

**FLTK Version**
 - Version: 1.4.0 master

**FLTK Configure / Build Options**
See runme.sh script.

**Operating System / Platform:**
 - OS: Windows
 - OS Version: Windows 11
 - OS Compiler: MSVC 2022

[build_fltk_rel_with_debinfo.zip](https://github.com/fltk/fltk/files/15123113/build_fltk_rel_with_debinfo.zip)

**Comment by ggarra13** [comment]
Okay.  False bug report.  The issue is that -DFLTK_DLL is not being set.  I believe there may be a way to pass this flag when doing:

find_package(FLTK CONFIG REQUIRED)

by adding a define to the fltk::fltk-shared windows target.

**Comment by Albrecht-S** [comment]
@ggarra13 Thanks for the report anyway. The current documentation says that the user program must define `FL_DLL`.

However, your suggestion is correct: we **_can_** set compile definitions on a target like `fltk::fltk-shared` so user code would automatically be compiled with that definition. At least in theory.

I'll think about this option.

**Comment by Albrecht-S** [comment]
@ggarra13 Can you please try the patch below? This should define the macro `FL_DLL` in MSVC (Visual Studio) projects that link to the shared FLTK library.
```diff
diff --git a/CMake/fl_add_library.cmake b/CMake/fl_add_library.cmake
index ebf9e28c0..c4e5f91c4 100644
--- a/CMake/fl_add_library.cmake
+++ b/CMake/fl_add_library.cmake
@@ -197,7 +197,7 @@ function(fl_add_library LIBNAME LIBTYPE SOURCES)
         OUTPUT_NAME_DEBUG   ${LIBNAME}_dlld
         OUTPUT_NAME_RELEASE ${LIBNAME}_dll
       )
-      target_compile_definitions(${TARGET_NAME} PRIVATE FL_DLL)
+      target_compile_definitions(${TARGET_NAME} PUBLIC FL_DLL)
     endif(MSVC)
   endif(LIBTYPE STREQUAL "SHARED")
 
```

Please let me know if this fixes the issue for your (example) project. Note that I'm not going to build your project myself (it's too large for me to check the details for such a tiny issue).

**Comment by ggarra13** [comment]
Seems to work fine on the simple test suite I gave you.  I'll test it on my real project later in the week.

**Comment by Albrecht-S** [comment]
Thanks for confirmation. I'm going to commit this and I will close this issue when done.

**Comment by Albrecht-S** [comment]
Fixed yesterday in commit 9cdd457382ea5da11c3dfa49320d920dc7ea7dfd
